composer/models/huggingface.py

```diff
@@ -96,13 +96,21 @@ def forward(self, batch):
         return output
 
     def loss(self, outputs, batch):
-        return outputs['loss']
+        if self.config.use_return_dict:
+            return outputs['loss']
+        else:
+            # loss is at index 0 in the output tuple
+            return outputs[0]
 
     def eval_forward(self, batch, outputs: Optional[Any] = None):
         output = outputs if outputs else self.forward(batch)
         if self.use_logits:
             self.labels = batch.pop('labels')
-            output = output['logits']
+            if self.config.use_return_dict:
+                output = output['logits']
+            else:
+                # logits are at index 1 in the output tuple
+                output = output[1]
```

Comment:
The assumption that loss and logits are at indices 0 and 1, respectively, depends on `labels` being in the batch. That seems like a safe assumption when extracting loss, but is there an edge case where the assumption doesn't hold for logits? I suppose `self.labels = batch.pop('labels')` acts like an `assert 'labels' in batch`, but if `outputs` is not `None` (i.e. it was precomputed and given as an input to `eval_forward`) can we be certain about the indexing?

I think this is probably an overly specific concern, so I'll give my approval. But it would be good to double check that logits at `output[1]` will always be a valid assumption.

Reply:
BTW indexing path won't be exercised for general HF models. It's only when someone specifically doesn't want dict outputs, this path will be exercised, i.e., by default it doesn't impact existing workflow at all. 

At least in the models I looked at (didn't look at all), this indexing works as expected.